src/test/java/uk/gov/homeoffice/digital/sas/timecard/producers/KafkaProducerServiceTest.java

```diff
@@ -17,60 +17,54 @@
 import org.springframework.kafka.core.KafkaTemplate;
 import org.springframework.kafka.support.SendResult;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.util.ReflectionTestUtils;
 import org.springframework.util.concurrent.ListenableFuture;
 import uk.gov.homeoffice.digital.sas.timecard.enums.KafkaAction;
-import uk.gov.homeoffice.digital.sas.timecard.model.KafkaEventMessage;
+import uk.gov.homeoffice.digital.sas.timecard.kafka.KafkaEventMessage;
+import uk.gov.homeoffice.digital.sas.timecard.kafka.producers.KafkaProducerService;
 import uk.gov.homeoffice.digital.sas.timecard.model.TimeEntry;
 
 @ExtendWith(SpringExtension.class)
-class KafkaProducerTimeEntryTest {
+class KafkaProducerServiceTest {
 
   @Mock
-  private KafkaTemplate<String, KafkaEventMessage> kafkaTimeEntryTemplate;
+  private KafkaTemplate<String, KafkaEventMessage<TimeEntry>> kafkaTimeEntryTemplate;
 
   @InjectMocks
-  private KafkaProducerTimeEntry kafkaProducerTimeEntry;
+  private KafkaProducerService<TimeEntry> kafkaProducerService;
 
   @Mock
-  private ListenableFuture<SendResult<String, KafkaEventMessage>> responseFuture;
+  private ListenableFuture<SendResult<String, KafkaEventMessage<TimeEntry>>> responseFuture;
 
   @ParameterizedTest
   @EnumSource(value = KafkaAction.class, names = {"CREATE", "UPDATE"})
   void sendMessage_actionOnTimeEntry_messageIsSentWithCorrectArguments(KafkaAction action) {
-    TimeEntry timeEntry = createTimeEntry();
+    ReflectionTestUtils.setField(kafkaProducerService, "topicName", "callisto-timecard");
```

Comment:
nit: can move `callisto-timecard` up to a instance level constant and reuse on line 64 too

Reply:
Updated